### ISSUE TITLE
http.ServerResponse: Cannot call a class constructor without |new|

### ISSUE THREAD

**Issue by sirenkovladd** [oven-sh/bun#4415]
### What version of Bun is running?

0.8.1+16b4bf341acc0f4804f0b6bdf5298c180cd00366

### What platform is your computer?

Linux 6.2.0-26-generic x86_64 x86_64

### What steps can reproduce the bug?

```js
const inject = require('light-my-request')

const dispatch = function (req, res) {
  const reply = 'Hello World'
  res.writeHead(200, { 'Content-Type': 'text/plain', 'Content-Length': reply.length })
  res.end(reply)
}

inject(dispatch, { method: 'get', url: '/' }, (err, res) => {
  console.log(res.payload)
});
```

short repro
```js
const http = require('http')
function Response(req) {
  http.ServerResponse.call(this, {});
}
console.log(new Response().constructor.name)
```

### What is the expected behavior?

```js
❯ node bun-http.ServerResponse.js
Hello World
```

less code
```js
❯ node bun-http.ServerResponse.js
Response
```

### What do you see instead?

```sh
❯ bun bun-http.ServerResponse.js 
621 | }
622 | var OriginalWriteHeadFn, OriginalImplicitHeadFn;
623 | 
624 | class ServerResponse extends Writable {
625 | 
626 |   constructor({ req, reply }) {
                              ^
TypeError: Cannot call a class constructor without |new|
      at ServerResponse (node:http:626:26)
      at new Response (<path>/bun-http.ServerResponse.js:15:2)
      at <path>/bun-http.ServerResponse.js:17:12
      at globalThis (<path>/bun-http.ServerResponse.js:17:43)
```

### Additional information

`fastify` library
https://github.com/fastify/light-my-request/blob/master/lib/response.js#L10

https://github.com/nodejs/node/blob/main/lib/_http_server.js#L195
https://github.com/oven-sh/bun/blob/bun-v0.8.1/src/js/node/http.ts#L952

**Comment by birkskyum** [comment]
This ServerResponse error seem similar to the ClientRequest error here:
- https://github.com/oven-sh/bun/issues/2297

**Comment by shaunmolloy** [comment]
Using bun v1.0.1 and a similar issue occurs using [snowflake-sdk](https://github.com/snowflakedb/snowflake-connector-nodejs), from the following code:

```ts
import * as snowflake from 'snowflake-sdk';

let connection: snowflake.Connection;

export async function getSnowflake(): Promise<snowflake.Connection> {
  if (connection) {
    return Promise.resolve(connection);
  }

  return new Promise((resolve, reject) => {
    const tmpConnection = snowflake.createConnection(params as ConnectionOptions);
    tmpConnection.connect((err) => {
      if (err) {
        reject(err);
        return;
      }
      debug('Connected to Snowflake');
      connection = tmpConnection;
      resolve(connection);
    });
  });
}
```

![2023-09-14-13-22-29](https://github.com/oven-sh/bun/assets/8168925/de2e19bf-38d4-4bd0-aa59-f891e7cef8f9)

**Comment by M-Gonzalo** [comment]
It's happening the same to me with an express server:

```js
const server = require('http').Server(app)
```

`TypeError: Cannot call a class constructor without |new|`

**Comment by isnifer** [comment]
Checked `light-my-request` code and figured out that it uses:
1. `setHeader` on ServerResponse and `bun test` returns an error: `TypeError: Cannot access invalid private field`
2. Next call in this class is `assignSocket` and it's not implemented in bun: `error: not implemented`

So, when you'll avoid an original error by changing:
```diff
-http.ServerResponse.call(this, req)
+const response = new http.ServerResponse(req)
+Object.assign(this, response)
```

You'll get errors that I described earlier.
In the nodejs after this patch — no changes, everything working as expected.

**Comment by paulleonartcalvo** [comment]
is there a workaround for this? Using fastify personally and encountering this error when using fastify.inject for unit testing.

**Comment by M-Gonzalo** [comment]
> is there a workaround for this?

For my particular case, I get by using
```js
const http = require('http')
const app = require('express')()
const server = http.createServer(app)
```

**Comment by paperclover** [comment]
the solution for this is to change ServerResponse in `http.ts` into an es5 function style class.

**Comment by isnifer** [comment]
Does [this PR](https://github.com/oven-sh/bun/pull/7485) helps to fix the issue? I've upgraded to 1.0.18 but the issue still persists.
`bun test` returns:

```sh
 1 | 'use strict'
 2 | 
 3 | const http = require('node:http')
 4 | const { Writable } = require('node:stream')
 5 | const util = require('node:util')
 6 |
 7 | const setCookie = require('set-cookie-parser')
 8 |
 9 | function Response (req, onEnd, reject) {
10 |   http.ServerResponse.call(this, req)
       ^
TypeError: Cannot call a class constructor without |new|
      at ServerResponse (node:http:672:27)
```

**Comment by M-Gonzalo** [comment]
I can't know until it's deployed and I can test it. @paperdave please don't close issues if you don't know they can actually be closed. I still see the same problem. Maybe when #7485 is integrated it will get fixed, I wouldn't know in the meantime.

**Comment by sirenkovladd** [comment]
@isnifer try `bun upgrade --canary` or wait for 1.0.19
This change was merged after 1.0.18

**Comment by M-Gonzalo** [comment]
> @isnifer try `bun upgrade --canary` or wait for 1.0.19 This change was merged after 1.0.18

It's still broken
![image](https://github.com/oven-sh/bun/assets/23223606/574b54ca-c4e1-4109-96b6-9a00b2e93148)

**Comment by sirenkovladd** [comment]
> It's still broken

Your issue is similar to, but beyond the scope of, this issue
since this is the http.Server class, and this issue specifies exactly about http.ServerResponse

however, I made a PR for your issue too https://github.com/oven-sh/bun/pull/7705
